services/venice-server/src/test/java/com/linkedin/venice/stats/AggServerQuotaUsageStatsTest.java

```diff
@@ -0,0 +1,66 @@
+package com.linkedin.venice.stats;
+
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+
+import com.linkedin.venice.throttle.TokenBucket;
+import io.tehuti.metrics.MetricsRepository;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+public class AggServerQuotaUsageStatsTest {
+  @Test
+  public void testAggServerQuotaUsageStats() {
+    MetricsRepository metricsRepository = new MetricsRepository();
+    AggServerQuotaUsageStats aggServerQuotaUsageStats = new AggServerQuotaUsageStats(metricsRepository);
+    String storeName = "testStore";
+    String storeName2 = "testStore2";
+    String readQuotaRequestedQPSString = "." + storeName + "--quota_request.Rate";
+    String readQuotaRequestedKPSString = "." + storeName + "--quota_request_key_count.Rate";
+    String readQuotaRequestedQPSString2 = "." + storeName2 + "--quota_request.Rate";
+    String readQuotaRequestedKPSString2 = "." + storeName2 + "--quota_request_key_count.Rate";
+    String totalReadQuotaRequestedQPSString = ".total--quota_request.Rate";
+    String totalReadQuotaRequestedKPSString = ".total--quota_request_key_count.Rate";
+    long batchSize = 100;
+    long batchSize2 = 200;
+    aggServerQuotaUsageStats.recordAllowed(storeName, batchSize);
+    aggServerQuotaUsageStats.recordAllowed(storeName, batchSize);
+    aggServerQuotaUsageStats.recordAllowed(storeName2, batchSize2);
+
+    double totalQPS = metricsRepository.getMetric(readQuotaRequestedQPSString).value()
+        + metricsRepository.getMetric(readQuotaRequestedQPSString2).value();
+    double totalKPS = metricsRepository.getMetric(readQuotaRequestedKPSString).value()
+        + metricsRepository.getMetric(readQuotaRequestedKPSString2).value();
+    Assert.assertEquals(metricsRepository.getMetric(totalReadQuotaRequestedQPSString).value(), totalQPS, 0.01);
```

Comment:
Should we also assert the expected raw values of these metrics?

Reply:
The rate metric calculation time window is very unpredictable, I tried it and it returns a different value depending on execution/call time.